File: Python/refer-offer/shu_zu_zhong_chu_xian_ci_shu_chao_guo_yi_ban_de_shu_zi_lcof.py
from typing import *

def majorityElement(nums: List[int]) -> int:
    key, value = nums[0], 1
    for i in range(1, len(nums)):
        if nums[i] == key:
            value += 1
        elif value == 0:
            key, value = nums[i], 1
        else:
            value -= 1
    
    return key

File: src/tmp/Java/src/com/dbc/algorithm/ReferOffer/ShuZuZhongChuXianCiShuChaoGuoYiBanDeShuZiLcof.java
package com.dbc.algorithm.ReferOffer;

public class ShuZuZhongChuXianCiShuChaoGuoYiBanDeShuZiLcof {
    public int majorityElement(int[] nums) {
        int key = nums[0], value = 1;
        for (int i = 1; i < nums.length; i++) {
            if (key == nums[i]) {
                value++;
            } else if (value == 0) {
                key = nums[i];
                value = 1;
            } else {
                value--;
            }
        }
        return key;
    }
}
